main.go

```diff
@@ -31,6 +31,8 @@ import (
 type server struct {
 	log            log.Logger
 	hardwareClient hardwareGetter
+	subscriptions  map[string]*subscription
+	subLock        sync.RWMutex
```

Comment:
mutexes should show up above the thing(s) its protects, and should be separated by a new line from other struct fields.